src/screens/Distribution/index.js

```diff
@@ -0,0 +1,239 @@
+import type { Currency } from "@ledgerhq/live-common/lib/types/currencies";
+import type { T } from "../../types/common";
+import TrackScreen from "../../analytics/TrackScreen";
+import { accountsSelector } from "../../reducers/accounts";
+import DistributionCard from "./DistributionCard";
+import LText from "../../components/LText";
+import type { DistributionItem } from "./DistributionCard";
+import {
+  counterValueCurrencySelector,
+  counterValueExchangeSelector,
+  currencySettingsSelector,
+  intermediaryCurrency,
+} from "../../reducers/settings";
+import CounterValues from "../../countervalues";
+import colors from "../../colors";
+import RingChart from "./RingChart";
+import CurrencyUnitValue from "../../components/CurrencyUnitValue";
+import globalSyncRefreshControl from "../../components/globalSyncRefreshControl";
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  distribution: AssetsDistribution,
+  counterValueCurrency: Currency,
+  t: T,
+};
+const List = globalSyncRefreshControl(FlatList);
+const mapStateToProps = createStructuredSelector({
+  distribution: state => {
+    const accounts = accountsSelector(state);
+    const counterValueCurrency = counterValueCurrencySelector(state);
+    const toExchange = counterValueExchangeSelector(state);
+    return getAssetsDistribution(accounts, (currency: Currency, value) => {
+      // $FlowFixMe
+      const currencySettings = currencySettingsSelector(state, { currency });
+      const fromExchange = currencySettings.exchange;
+      return CounterValues.calculateWithIntermediarySelector(state, {
+        from: currency,
+        fromExchange,
+        intermediary: intermediaryCurrency,
+        toExchange,
+        to: counterValueCurrency,
+        value,
+        disableRounding: true,
+      });
+    });
+  },
+  counterValueCurrency: counterValueCurrencySelector,
+});
+
+class Distribution extends PureComponent<Props, *> {
+  state = {
+    highlight: -1,
+  };
+  flatListRef: *;
+
+  static navigationOptions = {
+    title: i18next.t("distribution.header"),
+    headerLeft: null,
+  };
+
+  renderItem = ({ item, index }: { item: DistributionItem, index: number }) => (
+    <TouchableOpacity onPress={() => this.selectedKey(index)}>
+      <DistributionCard
+        item={item}
+        selectedKey={this.selectedKey}
+        highlighting={index === this.state.highlight}
+      />
+    </TouchableOpacity>
+  );
+
+  selectedKey = index => {
+    this.setState({ highlight: index });
+    if (typeof index === "number") {
+      this.flatListRef.scrollToIndex({ index }, true);
+    }
+  };
+
+  keyExtractor = item => item.id;
+
+  ListHeaderComponent = () => {
+    const { counterValueCurrency, distribution } = this.props;
+    const { highlight } = this.state;
+
+    return (
+      <View>
+        <View style={styles.header}>
+          <View style={styles.chartWrapper}>
+            <RingChart
+              side={180}
```

Comment:
what is this `side` number? is it the width of the thing? but it's not used as a width only in the data calculation?

if it is, probably rename to size and use it in the `style` of SVG.

it should should be calculated from the width of the screen to be responsive. (check what simulator width you were having and set like `Math.floor(deviceWidth * 180 / your_reference_size)`)